src/Build/BackEnd/Node/OutOfProcNode.cs

```diff
@@ -427,10 +428,7 @@ private NodeEngineShutdownReason HandleShutdown(out Exception exception)
         {
             if (_debugCommunications)
             {
-                using (StreamWriter writer = File.CreateText(String.Format(CultureInfo.CurrentCulture, Path.Combine(Path.GetTempPath(), @"MSBuild_NodeShutdown_{0}.txt"), Process.GetCurrentProcess().Id)))
-                {
-                    writer.WriteLine("Node shutting down with reason {0} and exception: {1}", _shutdownReason, _shutdownException);
-                }
+                CommunicationsUtilities.Trace("Shutting down with reason: {1},  and exception: {2}.", Process.GetCurrentProcess().Id, _shutdownReason, _shutdownException);
```

Comment:
Actually, have you looked at `CommunicationsUtilities.Trace`? It does the same thing, just writing to a different spot:
https://github.com/microsoft/msbuild/blob/master/src/Shared/CommunicationsUtilities.cs#L632-L639

Reply:
I figured as much. At least with `Trace` it'll be consistent with how we expect to debug node communication.